### ISSUE TITLE
Runtime CCSID detected as -2 and is invalid. Please change the CCSID in your user profile. Using fallback methods to access the IBM i file systems.

### ISSUE THREAD

**Issue by viruslogic** [codefori/vscode-ibmi#2094]
I have seen that there was already an issue with the same problem, but the solution provided was, doesn't work for me.

link: https://github.com/codefori/vscode-ibmi/issues/2065

I have a client where I am working correctly with the extension. But on another client it does not work for me.

OS version 7.3
Extension version: 2.10.1

I check if the SSH service is running: ps -ef | grep sshd

What I see is this:

qsecofr 1401 1 06-03-2024 14:31:25 - 00:00 pgm-sshd 

I stop the service: ENDTCPSVR SERVER(*SSHD)
I check the service again: It does not appear anymore.

I start the STRTCPSVR SERVER(*SSHD) service.
I check the service again and I see this again:

qsecofr 1416 1 06-03-2024 14:33:36 - 00:00 pgm-sshd  

When I access with VSCode I get the error...

On the other hand, I have found that in the as400 there are some jobs that eat a lot of percentage of the machine for many many time. Don't stop... I attach a screenshot

![image](https://github.com/codefori/vscode-ibmi/assets/10233590/b6c2c8b1-2bed-4c39-a60e-d2a805c7bd1b)

**Comment by chrjorgensen** [comment]
@viruslogic Please be aware that ending the SSH daemon (`ENDTCPSVR *SSHD`) does NOT end any jobs spawned from the SSH daemon! You have to kill / end any jobs manually. You can view the offending job info to see when it was started etc. to find the correct job to end.

Also note that we just published a new prerelease version 2.10.5, which solves a bug concerning the CCSID setting. Please install and re-test - and maybe run `Connect and Reload Server Settings` and report back.

**Comment by viruslogic** [comment]
@chrjorgensen Nothing, I upgraded to version 2.10.5 of the extension, ran Connect and Reload Server Settings and the result is the same. I've tried stopping and starting SSHD, I've made sure there were no jobs in the activity job... and nothing.

The jobs are still stuck. Even disconnecting vsCode from the server... eating a lot of cpu.

![image](https://github.com/codefori/vscode-ibmi/assets/10233590/9c952c6d-b844-4f2a-b131-ca2efcd09c3f)

The message of the extension tells me is: 

The system QCCSID is not set correctly. We recommend changing the CCSID on your user profile first, and then changing your system QCCSID. Runtime CCSID detected as -2 and is invalid. Please change the CCSID or default CCSID in your user profile. Using fallback methods to access the IBM i file systems.

**Comment by worksofliam** [comment]
@viruslogic What's the situation here? We don't control jobs and how they end. In most of our cases, job end successfully when due so.

What can we do for you here?

**Comment by viruslogic** [comment]
I finally managed to fix it. Checking the log of Code for IBM i, I saw that it gave errors when using some functions that apparently were not in the system, such as QDBSSUDF2. 

Pulling the thread I discovered that the system of my client was not up to date. Some cumulative PTFs were missing to be installed.

I asked for them to be applied and from then on it can work without problems.

As far as I am concerned, the issue can be closed.
